packages/beacon-node/src/chain/chain.ts

```diff
@@ -254,9 +256,9 @@ export class BeaconChain implements IBeaconChain {
             shufflingCache: this.shufflingCache,
             getHeadState: this.getHeadState.bind(this),
             bufferPool: new BufferPool(anchorState.type.tree_serializedSize(anchorState.node), metrics),
-            // datastore: new DbCPStateDatastore(this.db),
-            // TODO: add new flag
-            datastore: new FileCPStateDatastore(CHECKPOINT_STATES_FOLDER),
+            datastore: fileDataStore
+              ? new FileCPStateDatastore(CHECKPOINT_STATES_FOLDER)
```

Comment:
You should use a subpath of the main dirpath

Reply:
fixed in the latest commit